src/main/java/org/spongepowered/api/service/event/EventManager.java

```diff
@@ -34,13 +35,23 @@
     /**
      * Registers an object to receive {@link Event}s.
      *
-     * @param plugin A plugin instance
+     * @param plugin The plugin instance
      * @param obj The object
      * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin instance
      */
     void register(Object plugin, Object obj);
 
     /**
+     * Registers an event handler for a specific event class.
+     *
+     * @param plugin The plugin instance
+     * @param eventClass The event to listen to
+     * @param handler The handler to receive the events
+     * @param <T> The type of the event
+     */
+    <T extends Event> void register(Object plugin, Class<T> eventClass, EventHandler<? super T> handler);
```

Comment:
How do i specify the priority and ignoreCancelled flag?
Maybe we should pass in a `@Subscribe` (fake) instance or something similar (`EventHandlerOptions`).
That way we could make this method the main method and the method above (annotations) will (internally) use this method to register all found handlers.
We could also remove the above (annotations) method from this interface and move it to a fully implemented helper/util class. If you are interested i could write down some more details.